sklearn/preprocessing/tests/test_data.py

```diff
@@ -2377,3 +2378,190 @@ def test_power_transformer_lambda_zero():
+    X2 = np.array([[0, 2, 1], [0, 2, 1], [0, 2, 1], [1, 0, 2]])
+    y = np.array([0, 0, 1, 0])
+    cf_1 = CountFeaturizer().fit(X, y=y)
+    assert_array_equal(
+        cf_1.transform(X2),
+        np.array([[0, 2, 1, 1, 0], [0, 2, 1, 1, 0],
+                 [0, 2, 1, 1, 0], [1, 0, 2, 1, 1]]))
+
+
+def test_count_featurizer_diff_fit_multi_y():
+    # test count featurizer where fit is trained on something different
+    # than what we are transforming, now with multi-dimensional y
+
+    X = np.array([[0], [0], [1], [1]])
+    X2 = np.array([[1], [1], [0], [0]])
+    y = np.array([[0, 1], [0, 1], [0, 0], [0, 1]])
+    cf = CountFeaturizer().fit(X, y=y)
+    assert_array_equal(
+        cf.transform(X2),
+        np.array([[1, 2, 1, 1], [1, 2, 1, 1],
+                 [0, 2, 0, 2], [0, 2, 0, 2]]))
+
+    X2 = np.array([[1], [1], [1], [1]])
+    cf = CountFeaturizer().fit(X, y=y)
+    assert_array_equal(
+        cf.transform(X2),
+        np.array([[1, 2, 1, 1], [1, 2, 1, 1],
+                 [1, 2, 1, 1], [1, 2, 1, 1]]))
+
+
+def test_count_featurizer_inclusion():
+    # test with the inclusion parameter set
+
+    X = np.array([[0, 2, 1], [1, 0, 3], [1, 0, 2], [1, 0, 2]])
+    X2 = np.array([[0, 2, 1], [0, 2, 3], [1, 0, 5], [1, 1, 5]])
+    cf_inclusion_1 = CountFeaturizer(inclusion=[0])
+    cf_inclusion_2 = CountFeaturizer(inclusion=[0, 1, 2])
+    cf_inclusion_3 = CountFeaturizer(inclusion=[0, 1])
+    assert_array_equal(
+        cf_inclusion_1.fit_transform(X),
+        np.array([[0, 2, 1, 1], [1, 0, 3, 3], [1, 0, 2, 3], [1, 0, 2, 3]]))
+
+    assert_array_equal(
+        cf_inclusion_2.fit_transform(X),
+        np.array([[0, 2, 1, 1], [1, 0, 3, 1], [1, 0, 2, 2], [1, 0, 2, 2]]))
+
+    assert_array_equal(
+        cf_inclusion_3.fit_transform(X2),
+        np.array([[0, 2, 1, 2], [0, 2, 3, 2], [1, 0, 5, 1], [1, 1, 5, 1]]))
+
+
+def test_count_featurizer_y():
+    # test with the y parameter set
+
+    X = np.array([[0, 2, 1], [1, 0, 3], [1, 0, 2], [1, 0, 2]])
+    y = np.array([0, 0, 1, 0])
+    cf_1 = CountFeaturizer()
+    assert_array_equal(
+        cf_1.fit_transform(X, y=y),
+        np.array([[0, 2, 1, 1, 0], [1, 0, 3, 1, 0],
+                 [1, 0, 2, 1, 1], [1, 0, 2, 1, 1]]))
+
+
+def test_count_featurizer_y_inclusion():
+    # test with the inclusion and y parameter set
+
+    X = np.array([[0, 2, 1], [1, 0, 3], [1, 0, 2], [1, 0, 2]])
+    y = np.array([0, 0, 1, 0])
+    cf_inclusion_1 = CountFeaturizer(inclusion=[0])
+    cf_inclusion_2 = CountFeaturizer(inclusion=[0, 1, 2])
+    assert_array_equal(
+        cf_inclusion_1.fit_transform(X, y=y),
+        np.array([[0, 2, 1, 1, 0], [1, 0, 3, 2, 1],
+                 [1, 0, 2, 2, 1], [1, 0, 2, 2, 1]]))
+
+    assert_array_equal(
+        cf_inclusion_2.fit_transform(X, y=y),
+        np.array([[0, 2, 1, 1, 0], [1, 0, 3, 1, 0],
+                 [1, 0, 2, 1, 1], [1, 0, 2, 1, 1]]))
+
+
+def test_count_featurizer_multi_y():
```

Comment:
* test_count_featurizer_multi_y same question as (1) three additional values doesn't make sense to me.
Please explain in comments

Reply:
Does my comment on line 2527 clarify this? The first column in `y` contributes to 1 of the columns being added since it only takes 1 unique value. The second column in `y` contributes to 2 of the columns being added due to having two values 0 and 1.